lib/ACL/ACLStorageWrapper.php

```diff
@@ -94,7 +94,12 @@ public function rename($source, $target) {
 		if ($sourceParent === '.') {
 			$sourceParent = '';
 		}
-		return $this->checkPermissions($sourceParent, Constants::PERMISSION_DELETE) &&
+	        $targetParent = dirname($target);
+	        if ($targetParent === '.') {
+	        	$targetParent = '';
+	        }
+		return  ($sourceParent === $targetParent ||
+			$this->checkPermissions($sourceParent, Constants::PERMISSION_DELETE)) &&
```

Comment:
please add some parenthesis to clarify that the "parent check" only applies to the `DELETE` permissions check.

Reply:
No, I was just reading it wrong